### ISSUE TITLE
QEMU arm doesn't work

### ISSUE THREAD

**Issue by z3ntu** [postmarketOS/pmbootstrap#1017]
When running `./pmbootstrap.py qemu --arch=arm` (on a x86_64 host) I get the following error in the log.
```
(019941) [20:07:11] WARNING: Qemu is not using KVM and will run slower!
(019941) [20:07:11] % sudo chmod 666 /home/luca/.local/var/pmbootstrap/chroot_native/home/pmos/rootfs/qemu-vexpress.img
(019941) [20:07:11] NOTE: Run 'pmbootstrap qemu --image-size 2G' to set the system image size when you run out of space!
(019941) [20:07:11] Connect to the VM (telnet requires 'pmbootstrap initfs hook_add usb-shell'):
(019941) [20:07:11] * (ssh) ssh -p 2222 user@localhost
(019941) [20:07:11] * (telnet) telnet localhost 2224
(019941) [20:07:11] % qemu-system-arm -kernel /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-vexpress/boot/vmlinuz-postmarketos-stable -initrd /home/luca/.local/var/pmbootstrap/chroot
_rootfs_qemu-vexpress/boot/initramfs-postmarketos-stable -append "console=ttyAMA0,38400n8 console=tty1 PMOS_NO_OUTPUT_REDIRECT PMOS_FORCE_PARTITION_RESIZE" -m 1024 -netdev user,id=net0,host
fwd=tcp::2222-:22,hostfwd=tcp::2224-:24,net=172.16.42.0/24,dhcpstart=172.16.42.1 -show-cursor -dtb /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-vexpress/usr/share/dtb/vexpress-v2p-c
a9.dtb -M vexpress-a9 -sd /home/luca/.local/var/pmbootstrap/chroot_native/home/pmos/rootfs/qemu-vexpress.img -device virtio-net-device,netdev=net0 -vga virtio -display sdl,gl=on
WARNING: Image format was not specified for '/home/luca/.local/var/pmbootstrap/chroot_native/home/pmos/rootfs/qemu-vexpress.img' and probing guessed raw.
         Automatically detecting the format is dangerous for raw images, write operations on block 0 will be restricted.
         Specify the 'raw' format explicitly to remove the restrictions.
qemu-system-arm: Virtio VGA not available
```
The same error happens when running `./pmbootstrap qemu` on an ARM board.

**Comment by ollieparanoid** [comment]
Thanks for reporting! In pmbootstrap init, when selectin `dri-swrast` as Qemu driver, it's probably working again? If so, we shouldn't allow `dri-virtio` to be selected with `qemu-arm`.

**Comment by z3ntu** [comment]
![image](https://user-images.githubusercontent.com/3768500/34073767-baf21dcc-e2a1-11e7-983b-bcf88e833eb7.png)
That's how it looks with dri-swrast. Also ssh doesn't connect.
pmb log:
```
(002429) [20:42:23] % qemu-system-arm -kernel /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-vexpress/boot/vmlinuz-postmarketos-stable -initrd /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-vexpress/boot/initramfs-postmarketos-stable -append "console=ttyAMA0,38400n8 console=tty1 PMOS_NO_OUTPUT_REDIRECT PMOS_FORCE_PARTITION_RESIZE" -m 1024 -netdev user,id=net0,hostfwd=tcp::2222-:22,hostfwd=tcp::2224-:24,net=172.16.42.0/24,dhcpstart=172.16.42.1 -show-cursor -dtb /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-vexpress/usr/share/dtb/vexpress-v2p-ca9.dtb -M vexpress-a9 -sd /home/luca/.local/var/pmbootstrap/chroot_native/home/pmos/rootfs/qemu-vexpress.img -device virtio-net-device,netdev=net0 -display sdl,gl=on
WARNING: Image format was not specified for '/home/luca/.local/var/pmbootstrap/chroot_native/home/pmos/rootfs/qemu-vexpress.img' and probing guessed raw.
         Automatically detecting the format is dangerous for raw images, write operations on block 0 will be restricted.
         Specify the 'raw' format explicitly to remove the restrictions.
pulseaudio: set_sink_input_volume() failed
pulseaudio: Reason: Invalid argument
pulseaudio: set_sink_input_mute() failed
pulseaudio: Reason: Invalid argument
```

**Comment by ollieparanoid** [comment]
@MartijnBraam, I think you originally added arm Qemu support. Any idea?

**Comment by z3ntu** [comment]
Also no idea why 2 windows pop up... (both in x86_64->arm qemu and arm->arm qemu)

**Comment by ollieparanoid** [comment]
That doesn't happen for me - which distribution do you run? Right now we still depend on the host system's Qemu program (it would be better if we used the one from Alpine).

**Comment by z3ntu** [comment]
Arch Linux with `QEMU emulator version 2.10.1`

**Comment by PureTryOut** [comment]
Ooh yeah I noticed this earlier actually, `virtio` can not be used with virtual machine's of a different architecture than your host. They should probably fallback to `dri-swrast` instead of giving you the option to switch.

**Comment by z3ntu** [comment]
@PureTryOut I actually got the same error when trying to run the armhf qemu on an armv7 host (RPi3 - the processor is technically armv8 but kernel&userspace are armv7) so it can't just be the architecture difference.

**Comment by PureTryOut** [comment]
Isn't armhf basically armv6 though? If so, the VM you'd be running would still be different than your host architecture.

**Comment by z3ntu** [comment]
That's right...

**Comment by PureTryOut** [comment]
Which means it's not weird that `dri-virtio` still didn't work for you. That said, maybe it only works on x86_64, but that would be a bit weird to me.

**Comment by ollieparanoid** [comment]
I was wrong when I said the two screens don't happen to me. I just tested with `amd64`, but when running the `arm` version, it does have two screens. And I think I understand it now - it must be in Qemu's default profile for the `vexpress` machine we chose. We only did not notice it before, because the `gtk` gui was the default one, which allows you to toggle between all outputs (screens, serial, ...). The `sdl` gui can only show both screens side by side (and not serial output it seems).

So the PR I've made should fix everything, but you still need to choose the `--display=gtk` if you want it.
